tensorflow/python/ops/control_flow_ops_test.py

```diff
@@ -417,6 +419,16 @@ def testCondDuplicateArg2(self):
     with self.assertRaises(TypeError):
       control_flow_ops.cond(True, lambda: x, lambda: x, fn2=lambda: x)
 
+  def testCondWithGroupAndSummaries(self):
+    try:
+      writer = summary.create_file_writer('tb')
+      with writer.as_default(), summary.always_record_summaries():
+          op = control_flow_ops.cond(
+              random_ops.random_normal([]) >= 0,
+              lambda: control_flow_ops.group(summary.scalar('loss', 0.2)),
+              control_flow_ops.no_op)
+    except TypeError as e:
```

Comment:
I'd rather you have the test fail only with a statement of the form self.assert*, like self.assertAllEqual, self.assertRaises, etc.

In this one I'd validate some property of the op, like its number or outputs or sometihng like that.

Reply:
If I was to do something like that test would fail without this fix, exception TypeError would be raised on creating branch (it wouldn't even come to the assert that is validating op properties) and that's why I introduced try block so the test would fail in case that exception is raised. Is it better to do it with self.assertEqual?

Reply:
I'd rather write this test as something like

```
     writer = summary.create_file_writer('tb')
      with writer.as_default(), summary.always_record_summaries():
          op = control_flow_ops.cond(
              random_ops.random_uniform([]) >= -1,  # note: will always be true
              lambda: control_flow_ops.group(summary.scalar('loss', 0.2)),
              control_flow_ops.no_op)
      self.assertEqual(len(op.outputs), 0)
```

This way there's an assertion and if the error is raised the user will get a failure anyway

Reply:
I modified in the way you suggested, please check it out.